huaweicloud/services/dws/resource_huaweicloud_dws_ext_data_source.go

```diff
@@ -0,0 +1,497 @@
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("name", utils.PathSearch("name", extDataSource, nil)),
+		d.Set("type", utils.PathSearch("type", extDataSource, nil)),
+		d.Set("data_source_id", utils.PathSearch("data_source_id", extDataSource, nil)),
+		d.Set("user_name", utils.PathSearch("user_name", extDataSource, nil)),
+		d.Set("connect_info", utils.PathSearch("connect_info", extDataSource, nil)),
+		d.Set("description", utils.PathSearch("description", extDataSource, nil)),
+		d.Set("configure_status", utils.PathSearch("configure_status", extDataSource, nil)),
+		d.Set("status", utils.PathSearch("status", extDataSource, nil)),
+		d.Set("created_at", utils.PathSearch("created", extDataSource, nil)),
+		d.Set("updated_at", utils.PathSearch("data_source_updated", extDataSource, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func GetExtDataSource(cfg *config.Config, region string, d *schema.ResourceData, dsType string) (interface{}, error) {
+	var (
+		getDwsExtDataSourceHttpUrl = "v1.0/{project_id}/clusters/{cluster_id}/ext-data-sources"
+		getDwsExtDataSourceProduct = "dws"
+	)
+	getDwsExtDataSourceClient, err := cfg.NewServiceClient(getDwsExtDataSourceProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DWS Client: %s", err)
+	}
+
+	getDwsExtDataSourcePath := getDwsExtDataSourceClient.Endpoint + getDwsExtDataSourceHttpUrl
+	getDwsExtDataSourcePath = strings.ReplaceAll(getDwsExtDataSourcePath, "{project_id}", getDwsExtDataSourceClient.ProjectID)
+	getDwsExtDataSourcePath = strings.ReplaceAll(getDwsExtDataSourcePath, "{cluster_id}", fmt.Sprintf("%v", d.Get("cluster_id")))
+
+	getDwsExtDataSourcePath += fmt.Sprintf("?type=%s", dsType)
+
+	getDwsExtDataSourceOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders: map[string]string{
+			"Content-Type": "application/json;charset=UTF-8",
+		},
+		OkCodes: []int{
+			200,
+		},
+	}
+	getDwsExtDataSourceResp, err := getDwsExtDataSourceClient.Request("GET", getDwsExtDataSourcePath, &getDwsExtDataSourceOpt)
+
+	if err != nil {
+		return nil, err
+	}
+
+	getDwsExtDataSourceRespBody, err := utils.FlattenResponse(getDwsExtDataSourceResp)
+	if err != nil {
+		return nil, err
+	}
+
+	jsonPath := fmt.Sprintf("data_sources[?id=='%s']|[0]", d.Id())
+	rawData := utils.PathSearch(jsonPath, getDwsExtDataSourceRespBody, nil)
+
+	return rawData, nil
+}
+
+func resourceDwsExtDataSourceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	updateDwsExtDataSourceChanges := []string{
+		"reboot",
+		"user_name",
+	}
+
+	if d.HasChanges(updateDwsExtDataSourceChanges...) {
+		// updateDwsExtDataSource: update the DWS external data source.
+		var (
+			updateDwsExtDataSourceHttpUrl = "v1.0/{project_id}/clusters/{cluster_id}/ext-data-sources/{id}"
+			updateDwsExtDataSourceProduct = "aws"
```

Comment:
please check the product name is aws or dws?

Reply:
it's dws